### ISSUE TITLE
-a ebook-validate should exit with nonzero exit code if there are errors

### ISSUE THREAD

**Issue by slonopotamus** [asciidoctor/asciidoctor-epub3#215]
Subj, so it would become possible to use `-a ebook-validate` for non-interactive validation of produced files on CI or similar.

Also note that modern `epubcheck` (brought in by #214) has `--failonwarnings` switch, but by default it doesn't fail on warnings.

**Comment by mojavelinux** [comment]
That seems reasonable, I might even say necessary.

I wonder if there's a use case for running the validator for informational purposes without failing the command. We could use the value of `ebook-validate` to pass options to the validator. In those options could be a hint to fail if there are warnings. For example:

```
asciidoctor-epub3 -a ebook-validate=--fail-if-invalid book.adoc
```

However, if you'd like to go ahead and enable the suggested behavior by default and see how the users respond, you're free to take the approach. Perhaps later we might even add the inverse switch, `--ignore-warnings` to run it for information purposes only.

**Comment by slonopotamus** [comment]
The best way to fix this issue is to implement #176 first and then route epubcheck output into Asciidoctor logger.

**Comment by mojavelinux** [comment]
That's a neat idea.

**Comment by jnavila** [comment]
FYI, for Progit2,  we use the gem 'epubcheck-ruby' which, when invoked from command line fails when the ebook is invalid. See progit/progit2-pub's Rakefile:

```ruby
    puts "Converting to EPub..."
    exec_or_raise("bundle exec asciidoctor-epub3 #{progit_v}.asc")
    puts " -- Epub output at #{progit_v}.epub"

    exec_or_raise("epubcheck #{progit_v}.epub")
```

**Comment by slonopotamus** [comment]
asciidoctor-epub3 also uses `epubcheck-ruby` internally. It's just calling code that needs to be improved a bit.

**Comment by slonopotamus** [comment]
So!

After #176 was fixed, with Asciidoctor >= 1.5.7 you just need to `asciidoctor-epub3 --failure-level=ERROR -a ebook-validate <book.adoc>` and it will exit with non-zero exit status if encounters errors (I'm actually going to submit a testcase that checks this).

I think I don't want to attempt to extend this to older Asciidoctor releases.

@mojavelinux Do you think this is OK?

**Comment by mojavelinux** [comment]
> @mojavelinux Do you think this is OK?

Yes.

The purpose of supporting older releases is not to achieve feature parity. It's to smooth the transition to upgrade (instead of forcing people to upgrade because the build is broken).
